Automattic/jetpack
# Writing prompts: fix post meta not added when immediately publishing post

Fixes issue raised when reviewing https://github.com/Automattic/jetpack/pull/29214#discussion_r1125784683

## Proposed changes:

Changes from using the `transition_post_status` hook to the `wp_after_insert_post` hook when adding the `_jetpack_blogging_prompt_key` post meta.

This ensures that tags included with the post on create/update through the REST API are actually added to the post before we check for them when adding the prompt post meta.

### Other information:


## Jetpack product discussion

pe7F0s-uo-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Set up your wpcom sandbox for testing by applying the patch D102925-code and applying Jepack trunk to your sandbox with `bin/jetpack-downloader test jetpack trunk`.
- Set the JETPACK__SANDBOX_DOMAIN constant to your wpcom sandbox host, using a define statement on your development site, or Settings > Jetpack Constants on your jurassic.ninja site.
- Make sure your site is connected to WP.com (you will need to use local tunneling if using a local dev site PCYsg-GJ2-p2).
- Create a new post, insert a Writing Prompt block, add at least one more block, and immediately publish the post. Make sure you do not manually save a draft.
- Check to make sure the `_jetpack_blogging_prompt_key` post meta has been added with the prompt id.
